spec/lib/discourse_auth_cookie_spec.rb

```diff
@@ -0,0 +1,246 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe DiscourseAuthCookie do
+  fab!(:user) { Fabricate(:user) }
+
+  describe ".parse" do
+    it "does not attempt decryption or validations for v0 cookie" do
+      token = SecureRandom.hex
+      cookie = DiscourseAuthCookie.parse(token)
+      expect(cookie.token).to eq(token)
+    end
+
+    it "decrypts/validates signature for cookies that are < 32 chars" do
+      token = "asdad"
+      expect {
+        DiscourseAuthCookie.parse(token)
+      }.to raise_error(DiscourseAuthCookie::InvalidCookie)
+    end
+
+    it "decrypts/validates signature for v1 cookies" do
```

Comment:
I see v0 and v1 cookies being used around the PR. It might be good to document the definition somewhere.